tensorhive/core/utils/reservation_allowed.py

```diff
@@ -0,0 +1,78 @@
+from datetime import time, timedelta
+from sqlalchemy.orm.exc import NoResultFound
+from tensorhive.models.Resource import Resource
+
+
+def check_schedules(start_date, end_date, schedules):
```

Comment:
Again here, `check_schedules` doesn't tell what this method does. Additionally you could add a documentation comment explaining briefly what does this method do (and what exactly does it return - check_schedules that returns a date seems very confusing).

Furthermore, if you're going to follow my previous comment and throw it in a class, this method could be private (private methods don't exist per se in Python, but you could add double underscore (`__`) in front of method's name to "mark" it as private).